owncloud/core
# Enable all required sharing settings for sharees tests

## Description

## Related Issue

## Motivation and Context
Sometimes I am seeing the integration tests fail (on Jenkins and in my dev environment) with:
```
07:03:56   Scenario: Remote sharee for files               # /var/lib/jenkins/workspace/owncloud-core_core_PR-28970-KLQM2JCHDC72AQCUSH5EHC3SEWWK6KEXD6P3PRE6JRVOKZNTTWUQ/tests/integration/sharees_features/sharees_provisioningapiv2.feature:197
07:03:56 [Tue Sep 12 07:03:56 2017] ::1:50428 [404]: /ocs/v2.php/cloud/users/test
07:03:56 [Tue Sep 12 07:03:56 2017] ::1:50431 [200]: /ocs/v2.php/cloud/users
07:03:57 [Tue Sep 12 07:03:57 2017] ::1:50435 [200]: /ocs/v2.php/cloud/users/test
07:03:58 [Tue Sep 12 07:03:58 2017] ::1:50449 [200]: /ocs/v2.php/cloud/users/test
07:03:58 [Tue Sep 12 07:03:58 2017] ::1:50453 [404]: /ocs/v2.php/cloud/users/Sharee1
07:03:58 [Tue Sep 12 07:03:58 2017] ::1:50458 [200]: /ocs/v2.php/cloud/users
07:03:59 [Tue Sep 12 07:03:59 2017] ::1:50464 [200]: /ocs/v2.php/cloud/users/Sharee1
07:03:59 [Tue Sep 12 07:03:59 2017] ::1:50471 [200]: /ocs/v2.php/cloud/users/Sharee1
07:04:00 [Tue Sep 12 07:03:59 2017] ::1:50473 [404]: /ocs/v2.php/cloud/groups/ShareeGroup
07:04:00 [Tue Sep 12 07:04:00 2017] ::1:50476 [200]: /ocs/v2.php/cloud/groups
07:04:00 [Tue Sep 12 07:04:00 2017] ::1:50480 [200]: /ocs/v2.php/cloud/groups/ShareeGroup
07:04:00 [Tue Sep 12 07:04:00 2017] ::1:50482 [200]: /ocs/v2.php/cloud/users/test/groups
07:04:00 [Tue Sep 12 07:04:00 2017] ::1:50485 [200]: /ocs/v2.php/cloud/users/test/groups
07:04:00 [Tue Sep 12 07:04:00 2017] ::1:50488 [200]: /ocs/v2.php/cloud/users/test/groups
07:04:00     Given as an "test"                            # ShareesContext::asAn()
07:04:01 [Tue Sep 12 07:04:00 2017] ::1:50492 [200]: /ocs/v2.php/apps/files_sharing/api/v1/sharees?search=test%40localhost&itemType=file
07:04:01     When getting sharees for                      # ShareesContext::whenGettingShareesFor()
07:04:01       | search   | test@localhost |
07:04:01       | itemType | file           |
07:04:01     Then the OCS status code should be "200"      # ShareesContext::theOCSStatusCodeShouldBe()
07:04:01     And the HTTP status code should be "200"      # ShareesContext::theHTTPStatusCodeShouldBe()
07:04:01     Then "exact users" sharees returned is empty  # ShareesContext::thenListOfSharees()
07:04:01     Then "users" sharees returned is empty        # ShareesContext::thenListOfSharees()
07:04:01     Then "exact groups" sharees returned is empty # ShareesContext::thenListOfSharees()
07:04:01     Then "groups" sharees returned is empty       # ShareesContext::thenListOfSharees()
07:04:01     Then "exact remotes" sharees returned are     # ShareesContext::thenListOfSharees()
07:04:01       | test@localhost | 6 | test@localhost |
07:04:01       Failed asserting that two arrays are equal.
07:04:01       --- Expected
07:04:01       +++ Actual
07:04:01       @@ @@
07:04:01        Array (
07:04:01       -    0 => Array (...)
07:04:01        )
```


## How Has This Been Tested?
1) Manually uncheck "Allow apps to use the Share API", and the 2 Federated Cloud Sharing options "Allow users on this server to send shares to other servers" and "Allow users on this server to receive shares to other servers".
2) Run the integration test features in master:
```
sharees_features/sharees_provisioningapiv2.feature
sharees_features/sharees.feature
```
3) Confirm that the scenario at line 197 in each feature fails.
4) Switch to the branch with this code
5) Run the tests again and confirm they pass

## Types of changes

## Checklist:
